### PR TITLE
cleaning up internal naming convention

### LINKED ISSUE #529: Clean up internal naming
There's a lot of confusing naming in the internals.
Specifically, [almost everywhere](https://github.com/facebook/react-devtools/search?utf8=%E2%9C%93&q=element) `element` is used, it actually means `internalInstance`.

React has its own concept of elements which has nothing to do with `element` naming in DevTools, and it makes the code extra confusing.

We should rename all incorrect `element` names to `internalInstance`.
Happy to take contributions since it's not urgent but very useful :-)

### PR DESCRIPTION
**Pull request by hozefaj**
fixes #529

### CODE REVIEW

**Comment by gaearon** [comment]
Wow, that's super cool, thanks for sending it. Going to review soon.

File: agent/Agent.js [hunk 1]
```diff
@@ -210,18 +210,18 @@ class Agent extends EventEmitter {
       console.warn('unable to get the node for scrolling');
       return;
     }
-    var element = node.nodeType === Node.ELEMENT_NODE ?
+    var internalInstance = node.nodeType === Node.ELEMENT_NODE ?
```
Comment (gaearon) [inline agent/Agent.js]: This one is actually right.. somewhat. It refers to a DOM node. Let's call it `domElement`

File: agent/Agent.js [hunk 2]
```diff
@@ -336,19 +336,19 @@ class Agent extends EventEmitter {
...
+  getId(internalInstance: OpaqueNodeHandle): ElementID {
+    if (typeof internalInstance !== 'object' || !internalInstance) {
+      return internalInstance;
     }
-    if (!this.ids.has(element)) {
-      this.ids.set(element, guid());
-      this.reactElements.set(this.ids.get(element), element);
+    if (!this.ids.has(internalInstance)) {
+      this.ids.set(internalInstance, guid());
+      this.reactElements.set(this.ids.get(internalInstance), internalInstance);
```
Comment (gaearon) [inline agent/Agent.js]: Let's change `this.reactElements` to be `this.internalInstancesById`

File: agent/Agent.js [hunk 3]
```diff
@@ -336,19 +336,19 @@ class Agent extends EventEmitter {
...
-      return element;
+  getId(internalInstance: OpaqueNodeHandle): ElementID {
+    if (typeof internalInstance !== 'object' || !internalInstance) {
+      return internalInstance;
     }
-    if (!this.ids.has(element)) {
-      this.ids.set(element, guid());
-      this.reactElements.set(this.ids.get(element), element);
+    if (!this.ids.has(internalInstance)) {
+      this.ids.set(internalInstance, guid());
```
Comment (gaearon) [inline agent/Agent.js]: Let's change `this.ids` to be `this.idsByInternalInstances`

File: agent/inject.js [hunk 4]
```diff
@@ -21,10 +21,10 @@ module.exports = function(hook: Hook, agent: Agent) {
       agent.setReactInternals(id, helpers);
       helpers.walkTree(agent.onMounted.bind(agent, id), agent.addRoot.bind(agent, id));
     }),
-    hook.sub('root', ({renderer, element}) => agent.addRoot(renderer, element)),
-    hook.sub('mount', ({renderer, element, data}) => agent.onMounted(renderer, element, data)),
-    hook.sub('update', ({renderer, element, data}) => agent.onUpdated(element, data)),
-    hook.sub('unmount', ({renderer, element}) => agent.onUnmounted(element)),
+    hook.sub('root', ({renderer, internalInstance}) => agent.addRoot(renderer, internalInstance)),
```
Comment (gaearon) [inline agent/inject.js]: TODO before merging (for myself): this might break RN

File: backend/attachRendererFiber.js [hunk 5]
```diff
@@ -223,8 +222,8 @@ function attachRendererFiber(hook: Hook, rid: string, renderer: ReactRenderer):
     const alternate = current.alternate;
     if (alternate) {
       // TODO: relying on this seems a bit fishy.
-      const wasMounted = alternate.memoizedState != null && alternate.memoizedState.element != null;
-      const isMounted = current.memoizedState != null && current.memoizedState.element != null;
+      const wasMounted = alternate.memoizedState != null && alternate.memoizedState.internalInstance != null;
```
Comment (gaearon) [inline backend/attachRendererFiber.js]: Here `.element` is correct

File: backend/attachRendererFiber.js [hunk 6]
```diff
@@ -223,8 +222,8 @@ function attachRendererFiber(hook: Hook, rid: string, renderer: ReactRenderer):
...
+      const isMounted = current.memoizedState != null && current.memoizedState.internalInstance != null;
```
Comment (gaearon) [inline backend/attachRendererFiber.js]: Here too

File: backend/getData.js [hunk 7]
```diff
@@ -130,10 +130,10 @@ function getData(element: Object): DataType {
 }
 
 function setInProps(internalInst, path: Array<string | number>, value: any) {
-  var element = internalInst._currentElement;
```
Comment (gaearon) [inline backend/getData.js]: This method is correct, `element` should stay

File: backend/installGlobalHook.js [hunk 8]
```diff
@@ -84,7 +84,7 @@ function installGlobalHook(window: Object) {
       const mountedRoots = hook.getFiberRoots(rendererID);
       const current = root.current;
       const isKnownRoot = mountedRoots.has(root);
-      const isUnmounting = current.memoizedState == null || current.memoizedState.element == null;
+      const isUnmounting = current.memoizedState == null || current.memoizedState.internalInstance == null;
```
Comment (gaearon) [inline backend/installGlobalHook.js]: `element` should stay here

File: backend/integration/attach-0.14.js [hunk 9]
```diff
@@ -60,10 +60,10 @@ function setup(hook) {
   return handlers;
 }
 
-function wrapElement(hook, element) {
+function wrapElement(hook, internalInstance) {
```
Comment (gaearon) [inline backend/integration/attach-0.14.js]: `element` should stay here

File: backend/integration/attach-0.14.js [hunk 10]
```diff
@@ -60,10 +60,10 @@ function setup(hook) {
...
   var extras = attachRenderer(hook, 'abc', renderer);
   var node = document.createElement('div');
-  React.render(element, node);
+  React.render(internalInstance, node);
```
Comment (gaearon) [inline backend/integration/attach-0.14.js]: `element` should stay here

File: frontend/Highlighter/Overlay.js [hunk 11]
```diff
@@ -150,8 +150,8 @@ function findTipPos(dims, win) {
   return {top, left: dims.left + margin + 'px'};
 }
 
-function getElementDimensions(element) {
-  var calculatedStyle = window.getComputedStyle(element);
+function getElementDimensions(internalInstance) {
```
Comment (gaearon) [inline frontend/Highlighter/Overlay.js]: Let's rename it to `domElement`

**Comment by hozefaj** [comment]
@gaearon Changes made per your comment. 

I think Contributing.md should be updated about running `flow check` as well. I ran `npm run lint` but did not realize the flow errors until the CI job failed.

File: agent/inject.js (hunk 4 above)
Comment (hozefaj) [inline agent/inject.js]: I think other than this, which is not a update per se, all other review comments have been fixed.

**Comment by gaearon** [comment]
This looks great. Thank you so much!

**Pull request merged**